homeassistant/components/nina/binary_sensor.py

```diff
@@ -72,25 +72,28 @@ def __init__(
     @property
     def is_on(self) -> bool:
         """Return the state of the sensor."""
-        return len(self.coordinator.data[self._region]) > self._warning_index
+        if not len(self.coordinator.data[self._region]) > self._warning_index:
+            return False
+
+        data: NinaWarningData = self.coordinator.data[self._region][self._warning_index]
```

Comment:
We wouldn't need the explicit type annotation here with a correctly annotated coordinator above.

Reply:
Addressed in #79467